packages/plasma-web/src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import styled, { css, keyframes } from 'styled-components';
+
+import { ModalsContext } from './ModalsContext';
+import { ModalView, ModalViewProps } from './ModalView';
+
+export interface ModalProps extends ModalViewProps {
+    isOpen: boolean;
+}
+interface HidingProps {
+    isHiding?: boolean;
+}
+
+const wrapperHideAnimation = keyframes`
+    from {
+        opacity: 1;
+    }
+
+    to {
+        opacity: 0;
+    }
+`;
+const bodyShowAnimation = keyframes`
+    from {
+        opacity: 0;
+        transform: translate3d(0, -100%, 0);
+    }
+
+    to {
+        opacity: 1;
+        transform: translate3d(0, 0, 0);
+    }
+`;
+const bodyHideAnimation = keyframes`
+    from {
+        opacity: 1;
+    }
+
+    to {
+        opacity: 0;
+        transform: translate3d(0, -100%, 0);
+    }
+`;
+
+const StyledWrapper = styled.div<HidingProps>`
+    position: fixed;
+    top: 0;
+    left: 0;
+    right: 0;
+    bottom: 0;
+    z-index: 1;
+
+    display: flex;
+    align-items: center;
+    justify-content: center;
+
+    ${({ isHiding }) => css`
+        ${isHiding &&
+        css`
+            animation: 0.1s ${wrapperHideAnimation} ease-out;
+        `};
+    `}
```

Comment:
а не много `css` то?

```suggestion
    ${({ isHiding }) => isHiding && css`
        animation: 0.1s ${wrapperHideAnimation} ease-out;
     `}
```